### PR TITLE
feat: Add SheetSelection class for managing table cell selections wit…

### LINKED ISSUE #4906: Implement Shift+Arrow to select cell range
With a cell selected in a sheet, try the following keyboard shortcuts:

- <kbd>Shift</kbd>+<kbd>Up</kbd>
- <kbd>Shift</kbd>+<kbd>Down</kbd>
- <kbd>Shift</kbd>+<kbd>Left</kbd>
- <kbd>Shift</kbd>+<kbd>Right</kbd>

These keyboard shortcuts should grow/shrink a cell selection range. But they don't yet work.

In your browser console, notice a log message like this:

> Sheet selection resizing is not yet implemented

I had previously set up the event handlers for this shortcut but not finished implementing the selection mutation. This should be a pretty straightforward task for someone to pick up from here.

At the bottom of `SheetSelection.ts`, you can find the code location for this log statement. Finish implementing the `resized` method, as described in the comment. Play with Google Sheets to get a good understanding of how <kbd>Shift</kbd>+<kbd>Arrow</kbd> keys work. Pay particular attention to the (literal!) edge cases when the selection has reached the edge of the sheet.

### PR DESCRIPTION
**Pull request by MohammedTharikS**
Range restriction logic for SheetSelection and introduce InspectorSection and TableInspector components

Fixes #4906

This PR implements the [SheetSelection](cci:2://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:33:0-593:1) class to better manage table cell selections, specifically introducing logic for range restrictions and selection resizing. It adds the [resized](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:533:2-592:3) method to support Shift+Arrow key behaviors, ensuring selections mimic standard spreadsheet functionality while respecting column restrictions. It also introduces updates to the `InspectorSection` and `TableInspector` components.

**Technical details**

- Implemented [SheetSelection.ts](cci:7://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:0:0-0:0) with:
  - [resized(direction)](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:533:2-592:3): Handles resizing the selection range by moving the "anchor" cell (opposite to the active cell) in the specified direction.
  - Range restriction logic: Added [filterRestrictedCells](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:205:2-210:3), [filterRestrictedColumns](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:212:2-216:3), and [adjustRestrictedColumn](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:242:2-258:3) to ensure selections do not include columns that are range-restricted.
  - Updated selection creation methods ([ofColumnRange](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:151:2-188:3), [ofRowRange](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:126:2-149:3), [ofDataCellRange](cci:1://file:///c:/Users/thari/mathesar/mathesar_ui/src/components/sheet/selection/SheetSelection.ts:260:2-305:3)) to respect these restrictions.
- Added comprehensive tests in `SheetSelection.test.ts` covering various selection scenarios and edge cases.
- Updated `InspectorSection` component structure.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

### CODE REVIEW

File: mathesar_ui/src/components/sheet/selection/SheetSelection.ts [hunk 1]
```diff
@@ -547,10 +547,48 @@ export default class SheetSelection {
...
+    // Note that "moving the anchor" is just a mental model. In reality, we define
+    // a new selection between the *current* active cell and the *new* anchor position.
+    // The active cell remains the same, but the selection shape changes.
+    // Wait, my mental model in thought process was: Move the OPPOSITE end.
+    // If active is at A1 (TopLeft). Opposite is BottomRight.
+    // If I Shift+Down, I want to move the BottomRight DOWN.
+    // So I move the Anchor?
+    // In my variable naming: "anchor" = opposite.
+    // So "New Selection is between Active and (Moved Anchor)".
+    // Correct.
```
Comment (pavish) [inline mathesar_ui/src/components/sheet/selection/SheetSelection.ts:580]: If you're using AI, please understand, test, and review the code yourself before submitting a PR. At the very least, please read and remove AI generated comments.

**Review by pavish — COMMENTED** [review]
This code contains unrelated changes w.r.t button appearances and I'm not convinced that the author understands the logic here. 

I'm closing this PR.

@MohammedTharikS If you're raising another PR for the same, ensure that you remove unnecessary changes and review the code yourself first.
